src/types/activity.ts

```diff
@@ -58,12 +58,13 @@ export interface Activity {
   isMission: boolean;
   asMission: boolean;
   startTime: number;
+  endTime: number | null;

```

Comment:
Javascript `null` vs `undefined` is going to result in a land war somewhere some day.

If the object may be instantiated without the property, I'd prefer to use the syntax:
```suggestion
  endTime?: number;

```

Reply:
Huh. I swear I initially did that, using undefined rather than null, but it caused issues. Of course I try it now and it works a treat. Agreed that's preferred, so changing to that.